product-service/functions/addProduct/addProduct.js

```diff
@@ -0,0 +1,121 @@
+const { Client } = require('pg') ;
+
+
+const {
+  PG_HOST,
+  PG_PORT,
+  PG_DATABASE,
+  PG_USERNAME,
+  PG_PASSWORD,
+} = process.env;
+
+const dbOptions = {
+  host : PG_HOST,
+  port: +PG_PORT,
+  database: PG_DATABASE,
+  user: PG_USERNAME,
+  password: PG_PASSWORD,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+  connectionTimeoutMillis: 5000,
+};
+
+
+
+handleResponse = (products = {}, status = 200) => ({
+  headers: {
+    'Content-Type': 'application/json',
+    'Access-Control-Allow-Methods': '*',
+    'Access-Control-Allow-Origin': '*',
+  },
+  statusCode: status,
+  body: JSON.stringify(products),
+});
+
+module.exports.handler = async event => {
+
+  console.log(`[${new Date().toLocaleString()}]`);
+  console.log('[REQUEST]: add a product');
+  console.log(event.body);
+  
+
+  const client = new Client(dbOptions);
+
+
+  const { title, description, price, count } = JSON.parse(event.body)  ;
+  if (!title || price < 0 || count < 0) {
+    return handleResponse({
+      message: 'Bad request',
+    }, 400);
+  }
+  try {
+      console.log(12);
+    await client.connect();
+    await client.query('BEGIN');
+    await client.query(
+    `insert into products(title, description, price) values ($1, $2, ${price}) returning id`,
```

Comment:
сори, если честно не понимаю этот комент. зачем?
валидация+приведение типов должны идти до вставки в бд

Reply:
@AlinaLoz, I assumed that price was excluded from the prepared statement for the reason that it inserts a value as a string, and simply indicated that it could be used with an integer, nothing more.